src/main/kotlin/uk/gov/dluhc/printapi/messaging/service/PrintResponseProcessingService.kt

```diff
@@ -115,37 +110,10 @@ class PrintResponseProcessingService(
 
         certificateRepository.save(certificate)
 
-        sendEmailIfCertificateNotDelivered(newStatus, certificate, printResponse.requestId)
-    }
-
-    private fun sendEmailIfCertificateNotDelivered(newStatus: Status, certificate: Certificate, printResponseRequestId: String) {
-        if (newStatus == NOT_DELIVERED) {
-            try {
-                val request = with(certificate) {
-                    SendCertificateNotDeliveredEmailRequest(
-                        sourceReference = sourceReference!!,
-                        applicationReference = applicationReference!!,
-                        localAuthorityEmailAddresses = getLocalAuthorityEmailAddresses(gssCode!!),
-                        requestingUserEmailAddress = getRequestingUserEmailAddress(printRequests, printResponseRequestId)
-                    )
-                }
-                emailService.sendCertificateNotDeliveredEmail(request)
-            } catch (e: EmailNotSentException) {
-                logger.error(
-                    "failed to send Certificate Not Delivered email when processing ProcessPrintResponseMessage for " +
-                        "certificate [${certificate.id}] with requestId [$printResponseRequestId]: ${e.message}"
-                )
-            }
+        if (printResponse.statusStep == ProcessPrintResponseMessage.StatusStep.NOT_MINUS_DELIVERED) {
+            certificateNotDeliveredEmailSenderService.send(printResponse, certificate)
+        } else if (printResponse.status == ProcessPrintResponseMessage.Status.FAILED) {
+            certificateFailedToPrintEmailSenderService.send(printResponse, certificate)
```

Comment:
Nice and clear 👍  

I was about to suggest you could use a `when` statement instead, but unfortunately it's not the same variable being evaluated :(